quesma/clickhouse/parser.go

```diff
@@ -73,20 +73,27 @@ func JsonToColumns(namespace string, m SchemaMap, indentLvl int, config *ChTable
 			nested := JsonToColumns(nameFormatter.Format(namespace, name), nestedValue, indentLvl, config, nameFormatter)
 			resultColumns = append(resultColumns, nested...)
 		} else {
-			var fType string
+			var fTypeString string
 			if value == nil { // HACK ALERT -> We're treating null values as strings for now, so that we don't completely discard documents with empty values
-				fType = "Nullable(String)"
+				fTypeString = "Nullable(String)"
 			} else {
-				fType = NewType(value).String()
-				if !strings.Contains(fType, "Array") && !strings.Contains(fType, "DateTime") {
-					fType = "Nullable(" + fType + ")"
+				fType := NewType(value)
+
+				// handle "field":{} case (Elastic Agent sends such JSON fields) by ignoring them
+				if multiValueType, ok := fType.(MultiValueType); ok && len(multiValueType.Cols) == 0 {
+					continue
```

Comment:
It might be useful to generate log message in this case

Reply:
Added a log message here and also rebased on current main.